lightning/src/offers/offer.rs

```diff
@@ -425,19 +465,96 @@ tlv_stream!(OfferTlvStream, OfferTlvStreamRef, {
 	(22, node_id: PublicKey),
 });
 
+impl Bech32Encode for Offer {
+	const BECH32_HRP: &'static str = "lno";
+}
+
+impl FromStr for Offer {
+	type Err = ParseError;
+
+	fn from_str(s: &str) -> Result<Self, <Self as FromStr>::Err> {
+		Self::from_bech32_str(s)
+	}
+}
+
+impl TryFrom<ParsedMessage<OfferTlvStream>> for Offer {
+	type Error = ParseError;
+
+	fn try_from(offer: ParsedMessage<OfferTlvStream>) -> Result<Self, Self::Error> {
+		let ParsedMessage { bytes, tlv_stream } = offer;
+		let contents = OfferContents::try_from(tlv_stream)?;
+		Ok(Offer { bytes, contents })
+	}
+}
```

Comment:
I find this a bit more readable/concise, kinda jumping around the file right now: 
```
diff --git a/lightning/src/offers/offer.rs b/lightning/src/offers/offer.rs
index 5876acf6..a68a143f 100644
--- a/lightning/src/offers/offer.rs
+++ b/lightning/src/offers/offer.rs
@@ -400,7 +400,9 @@ impl TryFrom<Vec<u8>> for Offer {

        fn try_from(bytes: Vec<u8>) -> Result<Self, Self::Error> {
                let parsed_message = ParsedMessage::<OfferTlvStream>::try_from(bytes)?;
-               Offer::try_from(parsed_message)
+               let ParsedMessage { bytes, tlv_stream } = parsed_message;
+               let contents = OfferContents::try_from(tlv_stream)?;
+               Ok(Offer { bytes, contents })
        }
 }

@@ -477,16 +479,6 @@ impl FromStr for Offer {
        }
 }

-impl TryFrom<ParsedMessage<OfferTlvStream>> for Offer {
-       type Error = ParseError;
-
-       fn try_from(offer: ParsedMessage<OfferTlvStream>) -> Result<Self, Self::Error> {
-               let ParsedMessage { bytes, tlv_stream } = offer;
-               let contents = OfferContents::try_from(tlv_stream)?;
-               Ok(Offer { bytes, contents })
-       }
-}
-
 impl TryFrom<OfferTlvStream> for OfferContents {
        type Error = SemanticError;

Reply:
Yeah, good call.